lib/handsoap/service.rb

```diff
@@ -265,7 +274,7 @@ def dispatch(doc, action)
         fire_on_log_header { "HandSoap Response [#{dispatch_id}]: length: [#{@http_client.body_str.length}], HTTP-Status: [#{@http_client.response_code}], Content-Type: [#{@http_client.content_type}]" }
         fire_on_log_body { Handsoap.pretty_format_envelope(@http_client.body_str) }
 
-        soap_response = Response.new(@http_client.body_str, envelope_namespace)
+        soap_response = Response.new(@http_client, envelope_namespace)
```

Comment:
So, does this work for concurrent requests? `@http_client` can be common to multiple requests, so when we call `body_str` on it from within `Response` will we get the `body_str` for the right request? If curb caches these values by thread, then it's probably OK, but I'm not sure that's the case.